include/boost/dll/detail/ctor_dtor.hpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * ctor_dtor.hpp
+ *
+ *  Created on: 19.10.2015
+ *      Author: Klemens
+ */
+
+#ifndef INCLUDE_BOOST_DLL_DETAIL_CTOR_DTOR_HPP_
+#define INCLUDE_BOOST_DLL_DETAIL_CTOR_DTOR_HPP_
+
+#include <boost/config.hpp>
+#include <boost/function.hpp>
+#include <boost/dll/detail/mem_fn_cast.hpp>
+
+#if defined(BOOST_MSVC) || defined(BOOST_MSVC_VER)
+#include <boost/dll/detail/demangling/msvc.hpp>
+#else
+#include <boost/dll/detail/demangling/itanium.hpp>
+#endif
+
+namespace boost
+{
+namespace dll
+{
+
+namespace detail
+{
+
+template<typename Signature>
+struct ctor_t {};
+
+template<typename Class, typename ...Args>
+struct ctor_t<Class(Args...)>
+{
+	typedef void(Class::*mem_fn)(Args...) ;
+	typedef Class*(*pln_fn)(Args...) ;
+	typedef boost::function<void(Class * const, Args...)> standard;
```

Comment:
Can we use something instead `boost::function`? For example, just `void(Class * const, Args...)`?

`boost::function` applies some type erasure, andbrings unnecessary dependencies. Boost.DLL does not use `boost::function` in C++11 and newer modes.